taskcluster/scripts/pipeline/train_taskcluster.py

```diff
@@ -1,14 +1,125 @@
 #!/usr/bin/env python3
 
+import logging
+import os
 import os.path
+import requests
 import subprocess
 import sys
 
 TRAINING_SCRIPT = os.path.join(os.path.dirname(__file__), "train-taskcluster.sh")
+CONTINUATION_ARTIFACTS = {
+    "config.opustrainer.yml",
+    "config.opustrainer.yml.state",
+    "devset.out",
+    "model.npz",
+    "model.npz.best-bleu-detok.npz",
+    "model.npz.best-bleu-detok.npz.decoder.yml",
+    "model.npz.best-ce-mean-words.npz",
+    "model.npz.best-ce-mean-words.npz.decoder.yml",
+    "model.npz.best-chrf.npz",
+    "model.npz.best-chrf.npz.decoder.yml",
+    "model.npz.decoder.yml",
+    "model.npz.optimizer.npz",
+    "model.npz.progress.yml",
+    "model.npz.yml",
+    "opustrainer.log",
+    "train.log",
+    "valid.log",
+    "vocab.spm",
+}
+
+
+ARTIFACTS_URL = "{root_url}/api/queue/v1/task/{task_id}/runs/{run_id}/artifacts"
+ARTIFACT_URL = "{root_url}/api/queue/v1/task/{task_id}/runs/{run_id}/artifacts/{artifact_name}"
+# The argument number where pretrained model mode is expected.
+# This is 1-indexed, not 0-indexed, so it should line up with the argument
+# number this is fetched in in train-taskcluster.sh
+PRETRAINED_MODEL_MODE_ARG_NUMBER = 12
+# Nothing special about 17...just a number plucked out of thin air that
+# should be distinct enough to retry on.
+DOWNLOAD_ERROR_EXIT_CODE = 17
 
 
 def main(args):
```

Comment:
Thought: I guess ultimately we'll peel back all the layers to this onion and merge all of these separate scripts into a single training script that can just handle all of this, and write it all in python. I filed: #607 for this.

Reply:
Yeah. I'm not proud of introducing this extra layer of indirection, but I do intend to follow up when there's more time.